### DISCUSSION TITLE
OME-TIFF support in WSIReader

### DISCUSSION THREAD

**Discussion by vladpopovici** [Project-MONAI/MONAI#6474] [Q&A]
Hello,

I try to use WSIReader for accessing TIFF files that follow the [OME-TIFF specs](https://docs.openmicroscopy.org/ome-model/5.6.3/ome-tiff/). However, both the CUCIM and the TIFFFILE back-ends failed: CUCIM was not even able to read such files (_This format has more than one image with Subfile Type 0 so cannot be loaded!_) and TIFFFILE back-end found only 6 out of 10 levels. 
The OME-TIFF files were produced using the bioformats2raw + raw2ometiff "official" toolchain and I could access them correctly using my own code.
Since I would like to use the images without any further conversions, I wonder
1. whether there is a way of reading OME-TIFF files in MONAI that eluded me?
2. how to write my own back-end, in case the OME-TIFF is not yet supported? The list of back-ends seems to be built-in and I would rather avoid touching the MONAI code itself and write my code as a plug-in - at least as a starting point.

Thanks for any help and suggestions.

Vlad

**Comment by wyli** [comment]
Thanks for the question, it might be helpful to create a feature request at https://github.com/rapidsai/cucim for more info. 

To make a customized reader, currently it'll be inheriting `BaseWSIReader` or `CuCIMWSIReader`,  and implementing the appropriate methods.
